kornia/augmentation/augmentation.py

```diff
@@ -21,45 +21,45 @@
 
 
 class AugmentationBase(nn.Module):
-    def __init__(self, apply_fcn: Callable, return_transform: bool = False) -> None:
+    def __init__(self, return_transform: bool = False) -> None:
         super(AugmentationBase, self).__init__()
         self.return_transform = return_transform
-        self._apply_fcn: Callable = apply_fcn
-        self._params: Dict[str, torch.Tensor] = {}
 
     def infer_batch_shape(self, input: UnionType) -> torch.Size:
         return _infer_batch_shape(input)
 
-    def get_params(self, input_shape: torch.Size) -> Dict[str, torch.Tensor]:
+    def generate_parameters(self, input_shape: torch.Size) -> Dict[str, torch.Tensor]:
         raise NotImplementedError
 
-    def forward(self, input: UnionType, params: Optional[Dict[str, torch.Tensor]] = None) -> UnionType:  # type: ignore
+    def compute_transformation(self, input: torch.Tensor, params: Dict[str, torch.Tensor]) -> torch.Tensor:
```

Comment:
Do we really need to pass whole tensor here ? Maybe just shape is enough. Most of the times you might only need batch size, and in fact could be deduced from some of the params or can be passed f on generate parameters.

Reply:
Currently, our transform_matrix got dtype and device as well. If we want to keep the current behaviour, passing a Tensor is easier.

Reply:
ok, gotcha